### DISCUSSION TITLE
How to change the marker in geodataframe.plot()?

### DISCUSSION THREAD

**Discussion by florianjehn** [geopandas/geopandas#2784] [Q&A] [answered]
I have four geopandas dataframes which I plot in four subplots. I want to change the marker from "o" to something else. However, if I use the marker keyword it is simply ignored. How can I change the marker in geopandas.plot()

```
scenario="1"
fig, ax = plt.subplots(2, 2, figsize=(20,12), sharex=True, sharey=True)
ax = ax.flatten()
for i, crop in enumerate(crops):
    gdf_dict_gpd[crop+scenario].plot(
        ax=ax[i],
        column="value",
        cmap="viridis",
        legend=True,
        marker=".",
        markersize=0.002, 
        alpha=0.7,
        legend_kwds={'label': "Yield Reduction Ratio", "shrink": 0.45},
    )
    world.plot(ax=ax[i], color="none", edgecolor="black")
    if i == 0 or i == 2:
        ax[i].set_ylabel("Latitude")
    if i == 2 or i == 3:
        ax[i].set_xlabel("Longitude")
fig.subplots_adjust(wspace=0.5)
fig.subplots_adjust(wspace=0.5)
plt.savefig('LoI'+scenario+'.png', dpi=500)
```

**Comment by martinfleis** [comment, marked as answer]
Can you show the output of `geopandas.show_versions()`? Passing `marker="."` or marker="+" works on my side. Might be that with `markersize=0.002` they all are so tiny that look like a dot?

**Reply by florianjehn** [reply]
Show version returns:


```
SYSTEM INFO
-----------
python     : 3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:26:04) [GCC 10.4.0]
executable : /home/florian/local/anaconda3/envs/seaweed-growth-model/bin/python
machine    : Linux-5.4.0-137-generic-x86_64-with-glibc2.31

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.11.1
GEOS lib   : /home/florian/local/anaconda3/envs/seaweed-growth-model/lib/libgeos_c.so
GDAL       : 3.6.2
GDAL data dir: /home/florian/local/anaconda3/envs/seaweed-growth-model/share/gdal
PROJ       : 9.1.0
PROJ data dir: /home/florian/local/anaconda3/envs/seaweed-growth-model/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.11.1
numpy      : 1.23.1
pandas     : 1.5.3
pyproj     : 3.4.1
shapely    : 1.8.5.post1
fiona      : 1.8.22
geoalchemy2: None
geopy      : None
matplotlib : 3.5.1
mapclassify: 2.5.0
pygeos     : None
pyogrio    : None
psycopg2   : None
pyarrow    : None
rtree      : 1.0.1

```

I think I can reproduce the behavior. The smaller I make the markersize, the weirder the they behave. At markersize 1 they look like this: 

![image](https://user-images.githubusercontent.com/7868527/218084380-6437dee4-93db-4221-9082-3bc0857d82c4.png)


If I change it to 0.5 they look like this:

![image](https://user-images.githubusercontent.com/7868527/218084506-061a119d-7576-4893-84f6-2b4a6643e60e.png)

0.1:

![image](https://user-images.githubusercontent.com/7868527/218084593-b84199c6-ca01-478f-a7a8-faccebe92b10.png)


0.002:

![image](https://user-images.githubusercontent.com/7868527/218084683-d5b453b6-1018-4ae2-a418-7e08d2e1d01f.png)

**Reply by florianjehn** [reply]
And if I change the marker, it just randomly plots a subset as the marker I specified and another subset as "o"

![image](https://user-images.githubusercontent.com/7868527/218085008-be52ed06-4d95-4957-8563-d696bab8fb5e.png)

**Reply by martinfleis** [reply]
Is the behaviour the same with when using directly Matplotlib (I assume so)? If yes, it is nothing we can do here.

**Reply by florianjehn** [reply]
Just tested with plt.scatter() and it shows the same behavior. Too bad. But do you have an idea how I could just use larger markersize on those maps? Would I just have to make the figure larger?

**Reply by martinfleis** [reply]
Don't know, try asking over at matplotlib side.

**Reply by florianjehn** [reply]
Alright, thanks.
